[NETBEANS-5941] Add generic execute.task action to Gradle projects

lkishalmi opened the pull request:
My current work is based on the delivery branch, however I do not think this one would be required for 12.5.
This one would assign a real action to the GradleTask-s on the Navigator nodes.

I'm sorry I pushed a reformat on ActionProviderImpl.java...

sdedic commented on extide/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java:
```diff
@@ -500,14 +503,12 @@ Action forProject(Project p, FileObject fo) {
         private final Lookup lookup;
 
         @SuppressWarnings("OverridableMethodCallInConstructor")
         private CustomPopupActions(boolean onFile, boolean onGradleFile, FileObject fo) {
```
is `onGradleFile` parameter used ?

lkishalmi replied:
Well it is not used, but calculated earlier. This one is a copy+paste left over from the maven plugin. Probably have to check why we calculate that on the first place, also check that in the "original" maven plugin code as well.

sdedic reviewed (commented):
Sorry, I didn't quite get what's the functional difference between the previous 
```CustomActionMapping("name")```
and the new
```RunTaskActionMapping()``` 
?

lkishalmi commented on extide/gradle/src/org/netbeans/modules/gradle/actions/GradleTaskTokenProvider.java:
```diff
@@ -0,0 +1,80 @@
+       "taskPath",
+       "taskNames",
+       "taskPaths"
+    )));
```
Well, yes these are never meant to be used widely, though it would be good to find some place for the documentation of Replacement Token Providers. It would interest power users as well.

lkishalmi commented on extide/gradle/src/org/netbeans/modules/gradle/nodes/TaskNode.java:
```diff
@@ -45,6 +46,8 @@
     @StaticResource
     private static final String TASK_ICON = "org/netbeans/modules/gradle/resources/gradle-task.gif";
 
+    public static final String EXECUTE_TASK_ACTION = "execute.task"; //NOI18N
```
Good catch! I've put the constant in different places initially all around the source code. RunUtils might be not the final place.

lkishalmi commented on extide/gradle/src/org/netbeans/modules/gradle/api/execute/RunUtils.java:
```diff
@@ -90,6 +90,9 @@
     public static final String PROP_DEFAULT_CLI = "gradle.cli"; //NOI18N
 
+    /** Action to execute specific Gradle Tasks from the IDE (Eg. from Navigator). */
+    public static final String EXECUTE_TASK_ACTION = "execute.task"; //NOI18N
```
Yes, that's true, I'm aware of it.

lkishalmi commented on extide/gradle/src/org/netbeans/modules/gradle/actions/DefaultActionMapping.java:
```diff
@@ -146,6 +146,12 @@ public boolean equals(Object obj) {
+        return name + " [" + args + ']';
+    }
+
+    
```
Yes. It seems I was debugging the code...

lkishalmi commented:
Ad of ```RunTaskActionMapping()``` you are right, I changed the code first to get the mapping for ```execute.task```, then it included the replacement token, then I changed the code doing this unnecessary custom action mapping class. Thanks for the review! Really useful! I'm trying to get back to the game after a long pause...
